frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1343,6 +1348,16 @@ function ReaderFooter:onReaderReady()
     self:updateFooter()
 end
 
+function ReaderFooter:onReadSettings(config)
+    if not self.ui.document.info.has_pages then
+        local h_margins = config:readSetting("copt_h_page_margins") or
+            G_reader_settings:readSetting("copt_h_page_margins") or
+            DCREREADER_CONFIG_H_MARGIN_SIZES_MEDIUM
+        self.book_margin_left = h_margins[1]
+        self.book_margin_right = h_margins[2]
```

Comment:
So, that's all that was needed !? :) (I expected it would need much much work, so... well done :)

Given that you compute `math.floor((self.book_margin_left + self.book_margin_right)/2))` a few times above, you may as well compute it here, and just store the result as `self.book_margins_footer_width`.
And the same below in `ReaderFooter:onSetPageHorizMargins`.
And you would then use where appropriate:
`self.settings.progress_margin_width = Screen:scaleBySize(self.book_margins_footer_width)`
`return T(_("Same as book margins (%1)"), self.book_margins_footer_width)`